.github/workflows/tests.yml

```diff
@@ -24,3 +27,61 @@ jobs:
           && !contains(github.event.pull_request.labels.*.name, 'ci:run-full-test-suite')
           || false
          }}
+
+  upload-stable-deps:
+    needs: tests
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'push' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-tags: true
+          sparse-checkout: .github
+
+      - name: Prepare local repo
+        run: |
+          git fetch
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
+          if git ls-remote --exit-code origin "refs/heads/${{ env.DEPS_BRANCH }}"; then
+            git checkout "${{ env.DEPS_BRANCH }}"
+            git rebase origin/master
+          else
+            git checkout master
+            git checkout -b "${{ env.DEPS_BRANCH }}"
+          fi
+          rm -f .github/stable/*.txt

```

Comment:
# Question

Is there a danger that if one of the unit tests failed to upload the frozen requirements, they will be purged from the branch?

Reply:
not really because if it fails to upload, it _should_ fail the `tests` step above, and this shouldn't run at all. But stranger things have happened! anyway, it'd be on a branch and needs approval so it should be easy to catch.

Reply:
Good points! 🎉